streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -133,9 +132,9 @@ public StateStore getStateStore(String name) {
         if (node == null)
             throw new TopologyBuilderException("Accessing from an unknown node");
 
-        // TODO: restore this once we fix the ValueGetter initialization issue
-        //if (!node.stateStores.contains(name))
-        //    throw new TopologyBuilderException("Processor " + node.name() + " has no access to StateStore " + name);
+        if (!node.stateStores.contains(name)) {
```

Comment:
Can we add a test for this? Also i'm guessing the `//TODO` that was there before is no longer  valid?